artemis/src/test/java/com/artemis/EntitySubscriptionTest.java

```diff
@@ -214,6 +215,54 @@ public void subscription_remove_id_matches_entity() {
 		entity.deleteFromWorld();
 		world.process();
 	}
+	
+	@Test
+	public void systems_only_get_notified_of_changes_to_relevant_entities() {
```

Comment:
maybe rename this test now that we know the root cause? ensure_entity_subscription_integrity or something.